### DISCUSSION TITLE
Smooth term with random walk (RW2) in GAM using sdmTMB

### DISCUSSION THREAD

**Discussion by matteba** [sdmTMB/sdmTMB#188] [Q&A]
Hi, 
I'm trying to apply a smoother term for a environmental covariates in a GAM using the fuction model random walk (rw2) as implimented in INLA (see https://becarioprecario.bitbucket.io/inla-gitbook/ch-smoothing.html#sec:smoothterms). 
I was wondering if anybody came across this before or knows how it is possible to impliment it on sdmTMB. 
Thanks

**Comment by seananderson** [comment]
I'm pretty sure that's not currently possible. It might be possible if it's possible to include that in an mgcv `s()` call somehow. sdmTMB will let you have coefficients that follow a random walk or AR(1) process (`time_varying` formula and `time_varying_type` arguments) or use smoother functions with `s()` or `t2()` (including `s(..., by =)`, `k =`, 2D/3D smoothers, and setting knots via the `knots` argument). I haven't carefully followed everything going on with 'rw2' in INLA, but the wiggliness of the smoothers in sdmTMB is penalized via the magic of `mgcv::smooth2random`.
